hypershift-operator/controllers/nodepool/nodepool_controller_test.go

```diff
@@ -1017,9 +1209,158 @@ status: {}
 				},
 				Status: hyperv1.NodePoolStatus{},
 			},
-			tuningConfig: []client.Object{},
-			expect:       "",
-			error:        true,
+			tuningConfig:   []client.Object{},
+			tunedExpect:    "",
+			perfprofExpect: "",
+			error:          true,
+		},
+		//-------------------------------------------------------------------------
+		{
+			name: "gets a single valid PerformanceProfileConfig",
+			nodePool: &hyperv1.NodePool{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: namespace,
+				},
+				Spec: hyperv1.NodePoolSpec{
+					TuningConfig: []corev1.LocalObjectReference{
+						{
+							Name: "perfprofOne",
+						},
+					},
+				},
+				Status: hyperv1.NodePoolStatus{},
+			},
+			tuningConfig: []client.Object{
+				&corev1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "perfprofOne",
+						Namespace: namespace,
+					},
+					Data: map[string]string{
+						tuningConfigKey: perfprofOne,
+					},
+					BinaryData: nil,
+				},
+			},
+			tunedExpect:    "",
+			perfprofExpect: perfprofOneDefaulted,
+			error:          false,
+		},
+		{
+			name: "Should be at most one PerformanceProfileConfig per NodePool",
+			nodePool: &hyperv1.NodePool{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: namespace,
+				},
+				Spec: hyperv1.NodePoolSpec{
+					TuningConfig: []corev1.LocalObjectReference{
+						{
+							Name: "perfprofOne",
+						},
+						{
+							Name: "perfprofTwo",
+						},
+					},
+				},
+				Status: hyperv1.NodePoolStatus{},
+			},
+			tuningConfig: []client.Object{
+				&corev1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "perfprofOne",
+						Namespace: namespace,
+					},
+					Data: map[string]string{
+						tuningConfigKey: perfprofOne,
+					},
+				},
+				&corev1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "perfprofTwo",
+						Namespace: namespace,
+					},
+					Data: map[string]string{
+						tuningConfigKey: perfprofTwo,
+					},
+				},
+			},
+			tunedExpect:    "",
+			perfprofExpect: "",
+			error:          true,
+		},
+		{
+			name: "fails if a non existent PerformanceProfile is referenced",
+			nodePool: &hyperv1.NodePool{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: namespace,
+				},
+				Spec: hyperv1.NodePoolSpec{
+					TuningConfig: []corev1.LocalObjectReference{
+						{
+							Name: "does-not-exist",
+						},
+					},
+				},
+				Status: hyperv1.NodePoolStatus{},
+			},
+			tuningConfig:   []client.Object{},
+			tunedExpect:    "",
+			perfprofExpect: "",
+			error:          true,
+		},
+		{
+			name: "PerformanceProfiles and Tuned Configs could cohexists",
```

Comment:
what would happen if:
- One of the tuned profiles are not compatible with the performance one?
- One of the perf profiles overrides a tuned one?

which takes precedence when you have a profile for both cases?

Reply:
Ok, got it.

I think that the profiles will be parsed and moved to MachineConfig changes and processed by MCO, but maybe the precedence is important.